### PR TITLE
Print a shipping label: Yosemite layer

### PR DESCRIPTION
**Pull request by jaclync**
Part of #2166 

Based on Networking layer changes: https://github.com/woocommerce/woocommerce-ios/pull/3155

## Changes

Networking layer changes:
- Created `ShippingLabelRemoteProtocol`, and updated `ShippingLabelRemote` to implement this protocol 
- `ShippingLabelPrintData`: implemented `Equatable` for comparing two instances in unit tests

Yosemite layer changes:
- Created `ShippingLabelAction` and `ShippingLabelStore` to handle the action
- Created `MockShippingLabelRemote: ShippingLabelRemoteProtocol` and added unit tests for success and failure cases

## Testing
Just CI for now, the action isn't used in the app yet!

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Stores/ShippingLabelStore.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        switch self {
+        case .label:
+            return "label"
+        case .legal:
+            return "legal"
+        case .letter:
+            return "letter"
+        }
+    }
+}
```
Comment (pmusolino) [inline Yosemite/Yosemite/Stores/ShippingLabelStore.swift]: What do you think about removing this extension, and add `String` to the `ShippingLabelPaperSize` enum?

Like this:
```
public enum ShippingLabelPaperSize: String {
     case label
     case legal
     case letter
 }
```

File: Yosemite/YosemiteTests/Mockups/Networking/Remote/MockShippingLabelRemote.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/// Mock for `ShippingLabelRemote`.
+///
+final class MockShippingLabelRemote {
+    private struct PrintResultKey: Hashable {
+        let siteID: Int64
+        let shippingLabelID: Int64
+        let paperSize: String
+    }
+
+    /// The results to return based on the given arguments in `loadProduct`
```
Comment (pmusolino) [inline Yosemite/YosemiteTests/Mockups/Networking/Remote/MockShippingLabelRemote.swift]: nit: I think this is a typo -> `loadProduct`

**Review by pmusolino — APPROVED** [review]
LGTM :shipit: I just added two minor nits

File: Yosemite/Yosemite/Stores/ShippingLabelStore.swift (hunk 1 above)
Comment (jaclync) [reply Yosemite/Yosemite/Stores/ShippingLabelStore.swift]: I replaced the enum here with the one from the networking layer in: https://github.com/woocommerce/woocommerce-ios/pull/3155/commits/7fd4a2596885cd00b700ab2465af51f05102663d

There I conformed the enum to `RawRepresentable` following other enums in the networking layer, and the string raw value is an internal detail.

File: Yosemite/YosemiteTests/Mockups/Networking/Remote/MockShippingLabelRemote.swift (hunk 2 above)
Comment (jaclync) [reply Yosemite/YosemiteTests/Mockups/Networking/Remote/MockShippingLabelRemote.swift]: fixed in: https://github.com/woocommerce/woocommerce-ios/pull/3156/commits/88bc08df311eeb0c06fa17fc8f9e9c08599f2b9a

**Pull request merged**